dockstore-webservice/src/main/java/io/dockstore/webservice/languages/WDLHandler.java

```diff
@@ -259,6 +259,9 @@ public VersionTypeValidation validateEntrySet(Set<SourceFile> sourcefiles, Strin
                 // https://github.com/ga4gh/dockstore/issues/2139
                 validationMessageObject.put(primaryDescriptorFilePath, "Unknown methods were found, indicating that this may be a WDL 1.0 file. Currently Dockstore cannot parse WDL 1.0, so validation has been skipped. It is likely that the import processing and DAG generation will be broken.\n" + e.getMessage());
                 return new VersionTypeValidation(true, validationMessageObject);
+            } catch (StackOverflowError e) {
+                // This is niche and should be fixed properly with a new WDL parser.
+                throw new CustomWebApplicationException("Error parsing workflow. You may have a recursive import.", HttpStatus.SC_BAD_REQUEST);
```

Comment:
I could see this both ways, e.g., user is submitting a "bad" workflow, but this seems to be our problem, or at least the problem of WDL library that we're using, so I could see this instead being a 500 instead of a 400. Since we will hopefully fix it without the user needing to change anything, l lean towards 500, but don't feel super strongly against it if you keep it as a 400.